platform/default/src/mbgl/layermanager/layer_manager.cpp

```diff
@@ -65,6 +66,9 @@ LayerManagerDefault::LayerManagerDefault() {
 #if !defined(MBGL_LAYER_CUSTOM_DISABLE_ALL)
     addLayerType(std::make_unique<CustomLayerFactory>());
 #endif
+#if !defined(MBGL_LAYER_LOCATION_COMPONENT_DISABLE_ALL)
```

Comment:
This would enable LocationComponentLayer by default. Have you tried compiling with MBGL_RENDER_BACKEND_OPENGL=0?

Reply:
thanks, i did now. added guards in glfw that was the part that needed it